From 946d11336dc9ccd325e27ee8c2c93a8853b3e21b Mon Sep 17 00:00:00 2001
From: Yongxin Liu <yongxin.liu@windriver.com>
Date: Wed, 19 Jun 2024 15:57:51 +0000
Subject: [PATCH] qat17: remove kernel version check for PMD_SIZE/PMD_PAGE_SIZE

Due to the backport of kernel commit 82328227db8f ("x86/mm: Remove
P*D_PAGE_MASK and P*D_PAGE_SIZE macros") in linux-yocto v6.1, the
kernel version check for PMD_SIZE/PMD_PAGE_SIZE is not valid anymore.

Upstream-Status: Inappropriate [Code released in tarball form only]

Signed-off-by: Yongxin Liu <yongxin.liu@windriver.com>
---
 .../utilities/libusdm_drv/linux/kernel_space/qae_mem_drv.c    | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/quickassist/utilities/libusdm_drv/linux/kernel_space/qae_mem_drv.c b/quickassist/utilities/libusdm_drv/linux/kernel_space/qae_mem_drv.c
index ae96d3f..d4b4646 100644
--- a/quickassist/utilities/libusdm_drv/linux/kernel_space/qae_mem_drv.c
+++ b/quickassist/utilities/libusdm_drv/linux/kernel_space/qae_mem_drv.c
@@ -848,11 +848,7 @@ dev_get_user_page(struct file *fp, uint32_t cmd, unsigned long arg)
     {
        if (PageHuge(page))
        {
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6,2,0))
            if (user_mem_info.size != PMD_SIZE)
-#else
-           if (user_mem_info.size != PMD_PAGE_SIZE)
-#endif
            {
                mm_err("%s:%d dev_get_user_page: user_mem_info.size is not "
                       "equal to hugepage_size, ret=%d\n",
-- 
2.39.3

